docs/examples/model-energy.ipynb

```diff
@@ -0,0 +1,664 @@
+    "\n",
+    "It is important to keep in mind that actual power usage will be heavily dependent on the\n",
+    "specific details of the underlying software and hardware implementation. The numbers\n",
+    "provided by KerasSpiking should be taken as very rough estimates only, and they rely on\n",
+    "a number of assumptions:\n",
+    "\n",
+    "- **Device specifications**: In order to estimate the energy used by a model on a\n",
+    "particular device, we need to know how much energy is used per synaptic operation/neuron\n",
+    "update. We rely on published data for these numbers (see our sources for\n",
+    "[CPU/GPU/ARM](https://ieeexplore.ieee.org/abstract/document/7054508),\n",
+    "[Loihi](https://www.researchgate.net/publication/322548911_Loihi_A_Neuromorphic_Manycore_Processor_with_On-Chip_Learning),\n",
+    "and [SpiNNaker 1/2](https://arxiv.org/abs/1903.08941)). Energy numbers in practice can\n",
+    "differ significantly from published results.\n",
+    "- **Overhead**: We do not account for any overhead in the energy estimates (e.g., the\n",
+    "cost of transferring data on and off a device). We only estimate the energy usage of\n",
+    "internal model computations (synaptic operations and neuron updates). In practice,\n",
+    "overhead can be a significant contributor to the energy usage of a model.\n",
+    "- **Spiking implementation**: We assume that the model being estimated has been fully\n",
+    "converted to a spiking implementation when estimating the energy usage on a spiking\n",
+    "device (even if the input model has non-spiking elements). For example, if the model\n",
+    "contains ``tf.keras.layers.Activation(\"relu\")`` layers (non-spiking), we assume that on\n",
+    "a spiking device those layers will be converted to something equivalent to\n",
+    "``keras_spiking.SpikingActivation(\"relu\")``, and that any connecting layers (e.g.\n",
+    "``tf.keras.layers.Dense``) are applied in an event-based fashion (i.e., processing only\n",
+    "occurs when the input layer emits a spike). In practice, it is not trivial to map a\n",
+    "neural network to a spiking device in this way, and implementation details can\n",
+    "significantly affect energy usage. [Nengo](https://www.nengo.ai/nengo/) and\n",
+    "[NengoDL](https://www.nengo.ai/nengo-dl/) are designed to make this easier."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Using ModelEnergy\n",
+    "\n",
+    "The ``keras_spiking.ModelEnergy`` class provides the entry point for energy estimation.\n",
+    "It takes a Keras model as input, and computes relevant statistics for that model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# build an example model\n",
+    "inp = x = tf.keras.Input((28, 28))\n",
+    "x = tf.keras.layers.Flatten()(x)\n",
+    "x = tf.keras.layers.Dense(units=128)(x)\n",
```

Comment:
I'd suggest putting a conv layer in just to highlight the difference between connections and parameters. We might want to add a sentence or two just saying what "connections" and "neurons" represent. as well.